packages/console/src/components/Search/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FormEventHandler, KeyboardEventHandler, useState } from 'react';
+
+import SearchIcon from '@/icons/Search';
+
+import Button from '../Button';
+import TextInput from '../TextInput';
+import * as styles from './index.module.scss';
+
+type Props = {
+  defaultValue?: string;
+  onSearch?: (value: string) => void;
+};
+
+const Search = ({ defaultValue = '', onSearch }: Props) => {
+  const [inputValue, setInputValue] = useState<string>(defaultValue);
+
+  const handleSearchKeyPress: KeyboardEventHandler<HTMLInputElement> = (event) => {
+    if (event.key === 'Enter') {
+      onSearch?.(inputValue);
+    }
+  };
+
+  const handleSearchChange: FormEventHandler<HTMLInputElement> = (event) => {
+    setInputValue(event.currentTarget.value);
+  };
+
+  const handleClick = () => {
+    onSearch?.(inputValue);
```

Comment:
nit: Should we check if inputValue is empty?

Reply:
updated, PTAL